### PR TITLE
Added Contribution Guidelines fixes #146

### LINKED ISSUE #146: add Contribution Guidelines - CONTRIBUTING.md
add Contribution Guidelines section in README.md for GSSoC participants and other contributors.

### PR DESCRIPTION
**Pull request by Ankitanvesh**
Fixes #146 

#### Describe the changes you have made in this pr -
Added the contribution guidelines in another document CONTRIBUTING.md and linked it to the Readme.md file

### CODE REVIEW

**Comment by Ankitanvesh** [comment]
@armudgal I have reduced the size of CONTRIBUTING.md . Pls review it and let me know if there are any changes that you need me to do.

**Comment by Ankitanvesh** [comment]
Okay will do it

**Comment by armudgal** [comment]
Also, don't include CODE-OF-CONDUCT part cause we will have a seperate file for it.

**Comment by Ankitanvesh** [comment]
@armudgal pls review it and tell me if there is anything more that needs to be done.

**Comment by Ankitanvesh** [comment]
@armudgal pls review it I want to know if the changes made are good enough

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[Back to README](README.md)
+
+## Contributing
+
+When you are ready to start work on a issue:
+
+- Let us know by leaving a comment on the issue. (Also let us know later if you are no longer working on it.)
+
+- Once you are assigned the issue (or you have claimed the issue) only then proceed to making the Pull Request. This will help avoid multiple PR's pertaining to the same issue.
```
Comment (agarwalrounak) [inline CONTRIBUTING.md]: `PR's` 
Remove the apostrophe.

**Comment by agarwalrounak** [comment]
I think the development process shouldn't confine someone to a particular way of doing things. For example: `Create a branch whose name contains the github issue number`. Most people instead tend to name the branch depending on the changes they are making.

**Comment by Ankitanvesh** [comment]
Totally agree will do the requisite changes

**Comment by Ankitanvesh** [comment]
I have elaborated the first point a bit so that it provides the requisite freedom and at the same time if followed helps to improve readability.(

**Comment by Ankitanvesh** [comment]
@agarwalrounak I have done the changes pls review it

**Comment by agarwalrounak** [comment]
Don't really need a link to README.md from the CONTRIBUTING.md file! 

@armudgal will be able to guide better in this case.

**Comment by Ankitanvesh** [comment]
I agree but putting it there doesn't do much harm I guess. @armudgal  pls share your views on the issue.

**Comment by Ankitanvesh** [comment]
@armudgal pls review it

**Pull request merged**